odoo/orm/registry.py

```diff
@@ -621,6 +621,21 @@ def init_models(self, cr: Cursor, model_names: Iterable[str], context: dict[str,
             del self._foreign_keys
             del self._is_install
 
+    def check_null_constraints(self, cr: Cursor) -> None:
+        """ Check that all not-null constraints are set. """
+        cr.execute("SELECT table_name, column_name FROM information_schema.columns WHERE is_nullable = 'NO' AND table_schema = 'public'")
+        existing = set(cr.fetchall())
+
+        for field in (
+            field
+            for Model in self.models.values()
+            if Model._auto and not Model._abstract
+            for field in Model._fields.values()
+            if field.column_type and field.store and field.required
+            if (Model._table, field.name) not in existing
+        ):
+            _schema.warning("Missing not-null constraint on %s", field)
```

Comment:
Please avoid writing complex code when possible.
```suggestion
        for Model in self.models.values():
            if Model._auto and not Model._abstract:
                for field in Model._fields.values():
                    if (
                        field.column_type and field.store and field.required
                        and (Model._table, field.name) not in existing
                    ):
                        _schema.warning("Missing not-null constraint on %s", field)
```